nixos/modules/tasks/filesystems/zfs.nix

```diff
@@ -393,7 +393,9 @@ in
             };
             script = ''
               zpool_cmd="${packages.zfsUser}/sbin/zpool"
+              zfs_cmd="${packages.zfsUser}/sbin/zfs"
               ("$zpool_cmd" list "${pool}" >/dev/null) || "$zpool_cmd" import -d ${cfgZfs.devNodes} -N ${optionalString cfgZfs.forceImportAll "-f"} "${pool}"
+              ${optionalString cfgZfs.requestEncryptionCredentials "\"$zfs_cmd\" load-key -r \"${pool}\""}
```

Comment:
Did you actually test this?

Reply:
Just tested this, it works and loads the key (from file:///, not prompt). I just wonder why zfs-mount.service does not get started. (manual "systemctl start zfs-mount.service" works).